lore/io/__init__.py

```diff
@@ -165,9 +165,10 @@ def upload_object(obj, remote_path=None):
     if remote_path is None:
         raise ValueError("remote_path cannot be None when uploading objects")
     else:
-        with tempfile.NamedTemporaryFile() as f:
+        with tempfile.NamedTemporaryFile(delete=False) as f:
             pickle.dump(obj, f)
-            upload(f.name, remote_path)
+        upload_file(f.name, remote_path)
+        os.remove(f.name)
```

Comment:
Does this indicate that NamedTemporyFile does not sync it's write to disk until the file handle is disposed?

Reply:
> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

Indicates this is indeed a fix for Windows.

Reply:
Yes, the `NamedTemporyFile` does not seem to sync until the file handle is closed. Not only Windows, but I'm seeing the same behavior on MacOS as well. This should fix that